sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ResourceDataWriteProxy.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.ClientModel.Tests.Client.Models.ResourceManager.Resources;
+using System.Collections.Generic;
+using System.Text.Json;
+using ClientModel.Tests.ClientShared;
+
+namespace System.ClientModel.Tests.Client
+{
+    public class ResourceDataWriteProxy : IJsonModel<ResourceProviderData>
```

Comment:
It would be helpful to document the contract for creating a proxy type.  For example, for the MRW code to work correctly with a proxy, I believe the following must be true (feel free to correct me!):
1. The proxy type must implement `IPersistableModel<T>` (or `IJsonModel<T>`), where T is a type it's a proxy for
2. In the proxy type's Write implementation, it must pass the instance of the model it is writing -- this lives on options.ProxiedModel

Are there other pieces the implementation is relying on to function correctly that would make sense to include?